### DISCUSSION TITLE
Is it ok to leave ziggy out?

### DISCUSSION THREAD

**Discussion by magnilens** [tomasklaen/uosc#788] [General]
When I downloaded the new uosc, it was still triggering my Windows Defender. So I downloaded it into windows sandbox, removed the entire 'bin' folder, uploaded it to my google drive and downloaded it into my actual machine to install. 
It seems to be working, although I'm running into a problem with uosc freezing when I try to click on the buttons on the command bar. It doesn't happen all the time, but it seems to be happening often enough to concern me. Not sure if that's related.

**Comment by tomasklaen** [comment]
Even the binaries in 5.1.1 are triggering it? Damn it :( virustotal.com reports both microsoft and google are OK with them...

Well, ziggy is currently only used/called when searching or downloading subtitles in the `download-subtitles` menu, or to paste text from clipboard when using menu search input.

There's no reason its presence, or lack thereof, should in any way cause freezing. That will be some other issue.

**Reply by magnilens** [reply]
The problem could be within mpv itself then. That would be troublesome. In the meantime, it might be possible to bypass the freezing using keybindings. 
What would I need to put down in the input.conf to bind the subs, audio, chapter and playlist menus to keys?

**Reply by magnilens** [reply]
No good. I managed to bind the menus to keys, but if the buttons are frozen, even the keys don't work.

**Comment by christoph-heinrich** [comment]
When it freezes for you, can you check the console to see if there is an error?

**Reply by magnilens** [reply]
How do I check the console?

**Reply by christoph-heinrich** [reply]
You can create a keybinding with `x script-binding console/enable`.
The default keybinding is <kbd>`</kbd>

**Reply by magnilens** [reply]
Console is not showing anything new if the buttons are frozen. I also just noticed that if the timeline is called up when the buttons are frozen, it stays up and is frozen too.

**Reply by christoph-heinrich** [reply]
What do you mean it's not showing anything new?
I suspect uosc crashes, which is why it's UI doesn't update anymore and appears frozen.

**Reply by magnilens** [reply]
>What do you mean it's not showing anything new?

When I press the key to show the console, it shows a ton of text about the status of mpv. If I do something that mpv doesn't know how to respond to (like a double right click) it adds a line saying there's nothing bound to the action. But if uosc freezes when I click a button, there's no extra lines on the console saying there's anything gone wrong.

**Reply by christoph-heinrich** [reply]
ok alternative suggestion, start mpv from the command line with `--log-file=log.txt`, cause a freeze, and then send that log file.

**Reply by magnilens** [reply]
First logged execution and it crashed already. Here's the log.
[log.txt](https://github.com/tomasklaen/uosc/files/13328085/log.txt)

**Reply by christoph-heinrich** [reply]
You have two versions of uosc installed.
You need to delete `~~/scripts/uosc.lua` and `~~/scripts/uosc_shared/`

It's probably best if you follow the installation instructions in the README. That will delete all the old files and make sure you have the latest version installed.

**Reply by magnilens** [reply]
That can't be right, I did delete the `~~/scripts/uosc_shared/` folder already before I unzipped the new uosc into the mpv folder. But I deleted the 'uosc.lua' file and the crashes seem to be gone now. Opened half a dozen files and no problems. I'll let you know if they come back. If they don't, thanks for helping me figure that out.

**Comment by dyphire** [comment]
For those using mpv 0.40.0 and higher, I replaced ziggy's subtitle download and clipboard functions with curl and mpv clipboard properties and native commands in my personal fork: https://github.com/dyphire/uosc/commit/c1fb1dc76c0f2b3189277fbbb819877822aaaa71

The update function should also be possible via curl or wget, but I don't use this function so I'm not interested in it.
